collect_app/src/main/java/org/odk/collect/android/formentry/FormSessionRepository.kt

```diff
@@ -0,0 +1,45 @@
+package org.odk.collect.android.formentry
+
+import android.app.Application
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import org.odk.collect.android.javarosawrapper.FormController
+import org.odk.collect.androidshared.data.getState
+import org.odk.collect.shared.strings.UUIDGenerator
+
+interface FormSessionRepository {
+    fun create(): String
+    fun get(id: String): LiveData<FormController?>
+    fun set(id: String, formController: FormController)
+    fun clear(id: String)
+}
+
+class AppStateFormSessionRepository(application: Application) : FormSessionRepository {
```

Comment:
A Kotlin `object`? `AppStateFormSessionRepository` requires the current `Application` so it could be a Dagger `@Singleton` (although not sure that's worth doing), but not an `object` without making changes and giving it a static reference to the `Application`. `InMemFormSessionRepository` is only used in test, and you'd generally want a clean one for each test which would be more awkward if it was an `object`.

Reply:
Right my mistake.